src/functions-and-arrays.js

```diff
@@ -1,119 +1,294 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(a, b) {
+  // return a > b ? a : b;
+  if (a > b) {
+    return a;
+  } else {
+    return b;
+  }
+}
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
-function findLongestWord() {}
-
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot",
+];
 
+function findLongestWord(words) {
+  let longestWord = "";
+  if (words.length == 0) {
+    return null;
+  }
+  for (let i = 0; i < words.length; i++) {
+    if (words[i].length > longestWord.length) {
+      longestWord = words[i];
+    }
+  }
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(numbers) {
+  let sum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+    sum += numbers[i];
+  }
+  return sum;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
-
+function sum(mixedArr) {
+  let total = 0;
+  // if (typeof mixedArr !== "object") {
+  //   return null;
+  // }
+  for (let i = 0; i < mixedArr.length; i++) {
+    if (typeof mixedArr[i] === "string") {
+      total += mixedArr[i].length;
+    } else if (typeof mixedArr[i] === "number") {
+      total += mixedArr[i];
+    } else if (mixedArr[i] === true) {
+      total++;
+    } else if (mixedArr[i] === false) {
+      total = total + 0;
+    } else {
+      throw new Error("Unsupported data type sir or ma'am");
+    }
+  }
+  console.log(total);
+  return total;
+}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(numbersAvg) {
+  // let total = 0;
+  let avg = 0;
+  if (numbersAvg.length == 0) {
+    return null;
+  }
+  avg = sumNumbers(numbersAvg) / numbersAvg.length;
+  return avg;
+}
 
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace",
+];
 
-function averageWordLength() { }
+function averageWordLength(wordsArr) {
+  let total = 0;
+  let avg = 0;
+  if (wordsArr.length == 0) {
+    return null;
+  }
+  for (let i = 0; i < wordsArr.length; i++) {
+    total += wordsArr[i].length;
+    avg = total / wordsArr.length;
```

Comment:
You should calculate the average after the loop is finished calculating the total.